wallet-new/src/Cardano/Wallet/Server/Middlewares.hs

```diff
@@ -29,6 +36,21 @@ import           Pos.Launcher.Configuration (ThrottleSettings (..))
 withMiddlewares :: [Middleware] -> Application -> Application
 withMiddlewares = flip $ foldr ($)
 
+responseModifier :: Response -> Response
+responseModifier r
+    | responseStatus r == status415 =
+          responseBuilder status415
+          [ ("Content-Type", "application/json") ]
+          ( fromByteString .
+            toStrict .
+            encode $
+            UnsupportedMimeTypePresent "The API expects the Content-Type's main MIME-type to be 'application/json'"
+          )
+    | otherwise = r
+
+unsupportedMimeTypeMiddleware :: Middleware
+unsupportedMimeTypeMiddleware = modifyResponse responseModifier
```

Comment:
Since `responseModifier` is solely related to the above function, it would make sense to have it under a `where` clause. It doesn't make sense to use `responseModifier` as it is in any other context :+1: